fighters/packun/src/opff.rs

```diff
@@ -4,22 +4,153 @@ use super::*;
+        EFFECT(fighter, Hash40::new("sys_level_up"), Hash40::new("top"), -2, 10, 0, 0, 0, 0, 0.4, 0, 0, 0, 0, 0, 0, true);
+        if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 0 {
+            EFFECT_FOLLOW(fighter, Hash40::new("sys_grass"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1.5, false);
+        }
+        else if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 1 {
+            EFFECT_FOLLOW(fighter, Hash40::new("packun_poison_max"), Hash40::new("top"), 0, 15.5, 0, 0, 0, 0, 1.2, false);
+        }
+        else if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 2 {
+            EFFECT_FOLLOW(fighter, Hash40::new("sys_crown"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 0.9, false);
+        }
+        VarModule::off_flag(fighter.object(), vars::packun::instance::STANCE_INIT);
+    }
+}
+
+/// handle speed application
+unsafe fn check_apply_speeds(fighter: &mut smash::lua2cpp::L2CFighterCommon) {
+    
+    // handle speed application once
+    if VarModule::is_flag(fighter.object(), vars::packun::instance::STANCE_NEED_SET_SPEEDS) {
+        if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 0 {
+            apply_status_speed_mul(fighter, 1.0);
+        } else if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 1 {
+            apply_status_speed_mul(fighter, 0.88);
+        } else if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 2 {
+            apply_status_speed_mul(fighter, 0.75);
+        }
+        VarModule::off_flag(fighter.object(), vars::packun::instance::STANCE_NEED_SET_SPEEDS);
+    }
+
+    if fighter.status() != VarModule::get_int(fighter.object(), vars::packun::instance::STANCE_STATUS) {
+        //println!("Status is changing!");
+        VarModule::on_flag(fighter.object(), vars::packun::instance::STANCE_NEED_SET_SPEEDS);
+        VarModule::set_int(fighter.object(), vars::packun::instance::STANCE_STATUS, fighter.status());
+        //println!("new stance status: {}", VarModule::get_int(fighter.object(), vars::packun::instance::STANCE_STATUS));
+    }
+
+    // dash & momentum transfer speeds
+    if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 1 {
+        VarModule::set_float(fighter.object(), vars::common::instance::JUMP_SPEED_MAX_MUL, 0.88);
+
+        // if you are initial dash, slow them down slightly
+        if fighter.is_status(*FIGHTER_STATUS_KIND_DASH) {
+            let motion_vec = Vector3f {
+                x: -0.15 * PostureModule::lr(fighter.boma()) * (1.0 - (MotionModule::frame(fighter.boma()) / MotionModule::end_frame(fighter.boma()))),
+                y: 0.0, 
+                z: 0.0
+            };
+            KineticModule::add_speed_outside(fighter.boma(), *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
```

Comment:
Please don't use wind, dash should already be covered in the control energy status speed mul, no?